### PR TITLE
Change NotAuthorizedException's base class to GeneralSecurityException

### LINKED ISSUE #190: Use explicit security exceptions
Radar-auth authentication throw exceptions that extend from `RuntimeException`. This means there is no warning when they are not explicitly caught, and they can't be caught as one general exception. I'd propose to change the base class to [`GeneralSecurityException`](https://docs.oracle.com/javase/7/docs/api/java/security/GeneralSecurityException.html). Then they need to be caught explicitly, and they have a superclass that can be caught. @nivemaham @dennyverbeeck any thoughts?

### PR DESCRIPTION
**Pull request by dennyverbeeck**
Together with PR #191 this closes #190

### CODE REVIEW

File: radar-auth/src/main/java/org/radarcns/auth/authorization/RadarAuthorization.java [hunk 1]
```diff
@@ -26,11 +26,14 @@
 
     /**
      * Check if the user authenticated with the given token has the given permission. Not taking
-     * into account project affiliations.
+     * into account project affiliations. Throws a {@link NotAuthorizedException} if the supplied
```
Comment (blootsvoets) [inline radar-auth/src/main/java/org/radarcns/auth/authorization/RadarAuthorization.java]: This last sentence would not be necessary, it is documented with the `@throws` tag.

File: src/main/java/org/radarcns/management/service/SubjectService.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ private Role getProjectParticipantRole(ProjectDTO projectDTO) {
 
 
     @Transactional
-    public SubjectDTO updateSubject(SubjectDTO subjectDTO) throws IllegalAccessException {
+    public SubjectDTO updateSubject(SubjectDTO subjectDTO) {
```
Comment (nivemaham) [inline src/main/java/org/radarcns/management/service/SubjectService.java]: any idea why we had this exception earlier?

**Comment by dennyverbeeck** [comment]
The `ExceptionTranslator` class already had a handler for `NotAuthorizedException`, so it will still return a `FORBIDDEN` status code. Since the base class for `NotAuthorizedException` changed from unchecked type (`RuntimeException`) to a checked exception, we now need to explicitly add the throws declaration to the controller methods. But it does not affect the `ExceptionTranslator`.

File: src/main/java/org/radarcns/management/service/SubjectService.java (hunk 2 above)
Comment (dennyverbeeck) [reply src/main/java/org/radarcns/management/service/SubjectService.java]: Going back in the history, it was added in [this commit](https://github.com/RADAR-CNS/ManagementPortal/commit/df0608278860f9f0b27ae94cdae3386553cdf87a#diff-359c75b052347bcf9e55b4901495674eR109), but no other changes were made to the method in that commit. So I assume it was added when working on the method, and that the throwing code got removed again before the commit. In any case i don't see anything there now that would throw an `IllegalAccessException`.

**Comment by dennyverbeeck** [comment]
Indeed, the javadoc will be updated in a later PR i'm working on, related to #194 and #195, so i will go ahead with the merge for this PR now. Thanks!

**Pull request merged**